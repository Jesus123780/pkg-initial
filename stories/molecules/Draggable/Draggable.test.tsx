import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import { Draggable } from './index'

describe('Draggable', () => {
  it('renders correctly', () => {
    const { container } = render(<Draggable />)
    expect(container).toMatchSnapshot()
  })

  it('should start dragging when mouse is down', () => {
    const { getByTestId } = render(<Draggable />)
    const draggableElement = getByTestId('draggable-element')

    fireEvent.mouseDown(draggableElement)
    expect(draggableElement.style.position).toBe('fixed')
  })
})
