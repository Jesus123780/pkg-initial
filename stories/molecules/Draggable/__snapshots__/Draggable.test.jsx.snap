// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable renders correctly 1`] = `
<div>
  <div
    data-testid="draggable-element"
    style="width: 100%; position: fixed; left: 0px; top: 0px;"
  />
</div>
`;
